nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/json/AbstractJsonRowRecordReader.java

```diff
@@ -95,6 +100,41 @@ public AbstractJsonRowRecordReader(final InputStream in, final ComponentLog logg
         }
     }
 
+    protected AbstractJsonRowRecordReader(final InputStream in, final ComponentLog logger, final String dateFormat, final String timeFormat, final String timestampFormat,
+                                          final String skipToNestedJsonField) throws IOException, MalformedRecordException {
+
+        this(logger, dateFormat, timeFormat, timestampFormat);
+
+        try {
+            jsonParser = jsonFactory.createParser(in);
+            jsonParser.setCodec(codec);
+
+            if (skipToNestedJsonField != null) {
+                while (!jsonParser.nextFieldName(new SerializedString(skipToNestedJsonField))) {
+                    // go to nested field if specified
+                    if (!jsonParser.hasCurrentToken()) {
+                        throw new IOException("The defined skipTo json field is not found when processing json as NiFi record.");
+                    }
+                }
+                logger.debug("Skipped to specified json field [{}] when processing json as NiFI record.", skipToNestedJsonField);
+            }
+
+            JsonToken token = jsonParser.nextToken();
+            if (skipToNestedJsonField != null && !jsonParser.isExpectedStartArrayToken() && token != JsonToken.START_OBJECT) {
+                logger.debug("Specified json field [{}] to skip to is not found. Schema infer will start from the next nested json object or array.", skipToNestedJsonField);
+            }
```

Comment:
This check and debug log seems unnecessary, since the previous debug log already indicates where the parser is starting.